deposit/src/main/java/edu/unc/lib/deposit/fcrepo3/IngestDeposit.java

```diff
@@ -382,6 +397,58 @@ private void ingestObject(String ingestPid, boolean confirmExisting) throws Depo
 		// TODO increment ingestedOctets
 
 	}
+	
+	private boolean isDuplicateOkay(PID pid, String depositId) {
+		List<String> deposits = tsqs.fetchBySubjectAndPredicate(pid, Relationship.originalDeposit.toString());
+		
+		// Ensure that either the deposit ID of the file in the repository matches the current deposit
+		// or that there is no deposit id assigned, such as is the case for small ingests
+		if (deposits == null || deposits.size() == 0 || deposits.contains(this.getDepositPID().getURI())) {
```

Comment:
Could we use something like this logic to avoid having too many levels of scope?

```
if (there is a deposit record associated with this pid in the repository and it isn't the current deposit) {

  return false, because that means this pid definitely belongs to another deposit

}

if (there is not a staged file for this object) {

  return true
  
} else {

  if (we have a size for this object and the size of the staged file doesn't match) {
    return false
  }

  if (we have a checksum for this object and the checksum checksum of the staged file doesn't match) {
    return false
  }

  return true

}
```